bluemira/equilibria/coils/_grouping.py

```diff
@@ -0,0 +1,1088 @@
+            else symmetry_line
+        )
+        self._symmetry_matrix()
+
+    def _symmetry_matrix(self):
+        """
+        Symmetry matrix
+
+        .. math::
+
+            \\frac{1}{1 + m} \\left[ {\\begin{array}{cc}
+                                        1 - m^2 & 2m \\\\
+                                        2m & -(1 - m^2) \\\\
+                                      \\end{array} } \\right]
+        """
+        self._shift, grad = yintercept(self._symmetry_line)
+        grad2 = grad**2
+        mgrad2 = 1 - grad2
+        gradsq = 2 * grad
+        self.sym_mat = 1 / (1 + grad2) * np.array([[mgrad2, gradsq], [gradsq, -mgrad2]])
+
+    def _symmetrise(self) -> np.ndarray:
+        """
+        Calculate the change in position to the symmetric coil,
+        twice the distance to the line of symmetry.
+        """
+        cp = self._get_group_centre()
+        cp[1] -= self._shift
+        mirror = np.dot(self.sym_mat, cp.T)
+        mirror[1] += self._shift
+        return np.array([np.mean(self._coils[1].x), np.mean(self._coils[1].z)]) - mirror
+
+    @Circuit.x.setter
+    def x(self, new_x: float):
+        """
+        Set x coordinate of each coil
+        """
+        if isinstance(new_x, np.ndarray):
+            new_x = np.mean(new_x[0])
+        self._coils[0].x = self._coils[0].x + new_x - self._get_group_x_centre()
+        self._coils[1].x = self._coils[1].x - self._symmetrise()[0]
+
+    @Circuit.z.setter
+    def z(self, new_z: float):
+        """
+        Set z coordinate of each coil
+        """
+        if isinstance(new_z, np.ndarray):
+            new_z = np.mean(new_z[0])
+        self._coils[0].z = self._coils[0].z + new_z - self._get_group_z_centre()
+        self._coils[1].z = self._coils[1].z + self._symmetrise()[1]
+
+    def _get_group_x_centre(self) -> np.ndarray:
+        """Get the x centre of the first coil group"""
+        return np.mean(self._coils[0].x)
+
+    def _get_group_z_centre(self) -> np.ndarray:
+        """Get the z centre of the first coil group"""
+        return np.mean(self._coils[0].z)
+
+    def _get_group_centre(self) -> np.ndarray:
+        """Get the centre of the first coil group"""
+        return np.array([self._get_group_x_centre(), self._get_group_z_centre()])
+
+
+class CoilSet(CoilGroup):
+    """
+    CoilSet is a CoilGroup with the concept of control coils
+
+    A CoilSet will return the total volume and area of the coils
+    and the respective psi or field calculations can optionally sum over the coils
+    or only sum over the control coils.
+
+    By default all coils are controlled
+
+    Parameters
+    ----------
+    coils: Union[Coil, CoilGroup[Coil]]
+        The coils to be added to the set
+    control_names: Optional[List]
+        List of coil names to be controlled
+
+    """
+
+    def __init__(
+        self,
+        *coils: Union[Coil, CoilGroup[Coil]],
+        control_names: Union[bool, List] = True,
```

Comment:
What happens if this is `False`? Might it be better to treat `None` (i.e. no names given) as all coils, then an empty list to indicate no control coils (if that's allowed). Then the `Optional[List[str]]` in the docstring will be correct.

Reply:
I had that before but thought it could be confusing. If you've jumped to that too I will put it back

Reply:
Seems a reasonable API I think, but we should make sure it's spelled out in the docstring